action_plugins/get_deployer_job_provision_info.py

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/python
+# 
+# Copyright: (c) 2020, Johnathan Kupferer <jkupfere@redhat.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+#
+# This plugin is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import re
+import requests
+import yaml
+
+from ansible.errors import AnsibleError
+from ansible.module_utils.six import string_types
+from ansible.module_utils.parsing.convert_bool import boolean
+from ansible.plugins.action import ActionBase
+
+user_body_regex = re.compile(r'^user\.body:\s*');
+user_data_regex = re.compile(r'^user\.data:\s*');
+user_info_regex = re.compile(r'^user\.info:\s*');
+
+class ActionModule(ActionBase):
+    '''Print statements during execution and save user info to file'''
+
+    TRANSFERS_FILES = False
+    _VALID_ARGS = frozenset(('host','job_id','user','page_size','password','validate_certs'))
+
+    def run(self, tmp=None, task_vars=None):
+        self._supports_check_mode = True
+
+        if task_vars is None:
+            task_vars = dict()
+
+        result = super(ActionModule, self).run(tmp, task_vars)
+        result['_ansible_verbose_always'] = True
+        del tmp # tmp no longer has any effect
+
+        for required_arg in ('host', 'job_id', 'user', 'password'):
+            if not required_arg in self._task.args:
+                result['failed'] = True
+                result['error'] = required_arg + " is required"
+                return result
+
+        job_id = self._task.args.get('job_id')
+        host = self._task.args.get('host')
+        user = self._task.args.get('user')
+        password = self._task.args.get('password')
+        page_size = self._task.args.get('page_size', 100)
+        validate_certs = boolean(self._task.args.get('validate_certs', True), strict=False)
+
+        page = 1
+        url = f"https://{host}/api/v2/jobs/{job_id}/job_events/?page_size={page_size}"
+        provision_data = {}
+        provision_message_body = []
+        provision_messages = []
+
+        while True:
+          resp = requests.get(url, auth = (user, password), verify = validate_certs)
```

Comment:
Is retry+backoff taken care of by requests.get?
I think we want to deal with HTTP errors and eventually retry (gracefully), especially if we're going through all the events.

Reply:
The callback is preferred, but sometimes because anarchy is restarting or encounters an issue it isn't available to receive the callbacks. This functionality lets it catch up where it missed it.

Good call I think on adding some retries on the `requests.get` I will look in the morning.